docs/internal/expressions.md

```diff
@@ -0,0 +1,597 @@
+  Other::foo(); // Fully qualified, using type alias, this is fine.
+}
+```
+
+### Calling methods
+
+Calling a method in Verona is similar to most other languages.
+The name must be fully qualifiable (not necessarily fully qualified) to identify the type that implements it, either at compile time (for static calls) or at run-time (for dynamic calls).
+
+If the method has return values, the variable (named or temporary) that will hold the results must be of a compatible type, declared or inferred, with the rest of the code that uses it.
+
+Example:
+```ts
+my_function()
+{
+  // An object of type `MyType` (see above)
+  let a : MyType & mut = MyType;
+
+  // You can call methods via an object
+  // Here, the first argument becomes `a`
+  // Because we know the concrete type, this is a static call
+  let b = a.my_method();
+
+  // Or statically, with an explicit first argument
+  // This is identical to the one above
+  let c = MyType::my_method(a);
+
+  // Or use infix notation, equivalent to `MyType::===(U32, U32)`
+  let d : MyType & mut = MyType;
+  let (equals, result, error) = (d === 42);
+  // `equals` is a `Bool`, `result` is `MyType` and `error` is a `MaybeError`
+}
+```
+
+The notations used for calling functions are interchangeable and do not force semantics change (dynamic vs. static) calling, which will only be determined by being able to determine, at compile time, if the type is concrete or not.
+
+_Note: It's still under review if we'll use named parameters or forwarding arguments for calls._
+
+_Note2: Check the type document for a review on static vs. dynamic methods, constructors and destructors._
+
+## Equality
+
+Verona has two types of equality: by address and structural.
+
+### Address equality
+
+Address equality, compared with the keyword `is`, compares the address of the expressions, ie. their locations.
+Two variables can only have equal addresses if they are aliases to each other.
+
+Address equality implies type equality.
+Variables of non-compatible types are never equal.
+
+Example:
+```ts
+// Some variable
+var a : U32 & mut = 42;
+// An alias to the same address
+let b = a;
+// A new variable with the same value (but different address)
+var c : U32 & mut = 42;
+
+let eq = (a is b); // TRUE, `b` is an alias to `a`
+let ne = (a is c); // FALSE, `c` is a new variable
+
+// Types can be alias to sub-types
+interface Foo {}
+class Bar : Foo {}
+
+let d : Bar & mut = Foo;
+let e : Foo & mut = d;
+
+let eq = (d is e); // TRUE, `e` is an alias to `d` of a compatible type
+
+// Singleton types are always equal to themselves but not other singleton types
+class None {}
+class All {}
+
+let always_eq = (None is None); // TRUE
+let never_eq = (None is All); // FALSE
+```
+
+Because singleton types (other than numeric types) have no machine representation, they always compare equal on the same types, which means all variables of the same singleton type are aliases to each other.
```

Comment:
I think this is saying that singletons are singletons and comparing two pointers to the same object will always yield true.

Reply:
reworded